### DISCUSSION TITLE
[Documentation] Draw workplane on a different plane than x-y

### DISCUSSION THREAD

**Discussion by violet-fish** [dune3d/dune3d#67] [Q&A]
I'm trying to draw a workplane for a new body attached to a face of another body or even just on the x-z plane, but I can't seem to figure out how to rotate or offset the workplane.

I also can't find any documentation on workplanes other than that it's possible to unset a sketch from a work plane and set it to a new workplane

**Comment by violet-fish** [comment]
I just found the "Constrain workplane normal" tool, but I'm stuck trying to find how to draw a "u normal"

**Comment by ChrisJohnsen** [comment]
To constrain a workplane's normal you will need two lines in an earlier group. They will need to have a shared endpoint (constrained to be coincident).

Giving the workplane a name (in the Selection part of the UI while the workplane is click-selected) can help with seeing what is going on. You will be able to see how the name's text (not just the corner triangle) is transformed when you move the endpoints of the constraining lines (except for their endpoint coincidence, the lines themselves can be left mostly unconstrained; you will be able to move them even while the workplane's group is active to see their effects).

I believe that the "U" name comes from the use of U and V as names for the local axes of a plane that may be arbitrarily embedded in a larger 3D space. The U and V axes are the "local" X and Y axes of the plane. These alternate names are used to know that we are talking about the "local" axes, not the global X or Y axes. The workplane's normal is a vector perpendicular to the plane itself (i.e. it points "out of" the plane); in Dune 3D, it is represented by a short line that extends away from the workplane's origin.

I'll name the endpoints A, B, and C: A is the shared endpoint, B is the other end of the first line.

    C
     \
      A--B  (U, first selected)

The A, B, and C points determine most of the orientation of the plane: geometrically, [three points determine a plane][plane]. Then, I think of the U vector (the first selected line: AB) as pointing in the same direction as, and parallel to, the positive horizontal axis of the workplane (its "local X" axis). The second line determines the direction of the vertical axis (or equivalently which direction the workplane "faces", as indicated by the normal), but less directly: the second line's actual direction doesn't matter, only which "side" of the AB line the C point lies.

[plane]: https://en.wikipedia.org/wiki/Euclidean_planes_in_three-dimensional_space#Determination_by_contained_points_and_lines

The workplane normal constraint seems to be set up so that (at the time it is applied) the workplane will "face" the camera (i.e. the name may end up appearing rotated, but will not be mirrored). After that, the position of point C controls which direction the workplane faces (moving C to the "other side" of AB flips the plane over, and (equivalently) inverts the direction of the V/"local Y" axis). So, workplanes constrained with the same lines, selected in the same order, can end up facing in opposite directions if the camera was viewing (approximately) opposite directions when the constraint was applied.

**Reply by magdesign** [reply]
thanks for the explanation, which helped, but,
my workplane is now somewhere, how can i attach it to the surface of the object?
<img width="996" height="752" alt="dune3d" src="https://github.com/user-attachments/assets/87c42dd2-b8ca-417e-ab8b-fceaf83b080a" />

following the [tutorial](https://docs.dune3d.org/en/1.3/simple-part.html)

**Reply by carrotIndustries** [reply]
>  my workplane is now somewhere, how can i attach it to the surface of the object?

What part of that isn't answered by the tutorial or the reply above?

**Reply by magdesign** [reply]
after several tries i finally got it.
do you mind if i take the time to make some additions in your **Designing a simple part** tutorial docs file, so its maybe easier to understand for total noobs ?
(i like to write docs...😅)

**Comment by magdesign** [comment]
Note: I'm actually  updating the tut right now 😱
